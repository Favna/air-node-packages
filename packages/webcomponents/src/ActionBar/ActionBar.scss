@import '../styles/vars';
@import '../styles/breakpoints';

.action {
  &Bar {
    background-color: $yellow20;
    height: 5rem;
    line-height: 5rem;
  }

  &Header {
    font-size: 2.2rem;

    @include md {
      font-size: 1.6rem;
    }

    @include smallLaptop {
      font-size: 1.6rem;
    }

    @include sm {
      font-size: 1.1rem;
    }
  }

  &Grid {
    &Left {
      height: 100%;
      padding-left: 2rem;

      @include sm {
        padding-left: 1rem;
      }
    }
    &Right {
      padding-right: 2rem;

      @include sm {
        padding-right: 1rem;
      }
    }
  }
}
