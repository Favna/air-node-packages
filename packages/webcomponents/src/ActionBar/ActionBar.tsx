import Grid from '@material-ui/core/Grid/Grid';
import css from './ActionBar.scss';
import Button from '../Button/Button';
import classnames from 'classnames';
import React, { FC, Fragment } from 'react';
import { customCss, dataQa } from '../typings';

export type ActionBarProps = {
  /** The title to show in the ActionBar */
  title: string;
  /** Whether an action button should be displayed in the ActionBar */
  shouldHaveButton?: boolean;
  /** The text that should be shown in the button */
  buttonLabel?: string;
  /** data-qa tag to apply to the search bar and input element */
  'data-qa'?: dataQa
  /** Custom CSS classes to pass to the button */
  customclasses?: customCss;
  /** The action that should be invoked when the button is clicked */
  buttonAction?: (...args: any[]) => any;
};

const ActionBar: FC<ActionBarProps> = props => {
  return (
    <div className={classnames('navigation-bar', css.actionBar)} data-qa={props['data-qa']}>
      <Grid container direction='row' justify='space-between' alignItems='center' style={{ height: '100%' }}>
        <Grid item key={1} xs={6}>
          <h1 data-qa='action-bar-title' className={classnames(css.actionBarHeader)}>
            {props.title}
          </h1>
        </Grid>
        <Grid item key={2} xs={6}>
          {props.shouldHaveButton ?
            <Button
              data-qa='action-bar-button'
              variant='contained'
              color='primary'
              label={props.buttonLabel}
              onClick={() => props.buttonAction ? props.buttonAction() : null}
            />
            : <Fragment />
          }
        </Grid>
      </Grid>
    </div>
  );
};

export default ActionBar;
