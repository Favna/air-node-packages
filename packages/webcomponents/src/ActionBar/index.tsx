import { Typography, TypographyProps, Box } from '@material-ui/core';
import Grid from '@material-ui/core/Grid/Grid';
import classnames from 'classnames';
import React, { memo, ReactNode, useMemo } from 'react';
import { When } from 'react-if';
import Button from '../Button';
import css from './ActionBar.scss';

export interface ActionBarProps {
  /** The title to show in the ActionBar */
  title: ReactNode;
  /**
   * Whether an action button should be displayed in the ActionBar
   * @default false
   */
  shouldHaveButton?: boolean;
  /**
   * Whether the button should be disabled
   * Only relevant when `shouldHaveButton` is set to true
   * @default false
   */
  shouldDisableButton?: boolean;
  /** The text that should be shown in the button */
  buttonLabel?: string;
  /** Typography variant for the header to take */
  typographyVariant?: TypographyProps['variant'];
  /** Custom CSS classes to pass to the title */
  customclasses?: string | string[];
  /** Data-qa tag to apply action bar bounding box */
  'data-qa'?: string;
  /** The action that should be invoked when the button is clicked */
  buttonAction?: (...args: any[]) => any;
}

/**
 * Creates an action bar using pre-defined Rijkswaterstaat styling
 * @param props Props to pass to the actionbar
 * @example
 * ```jsx
 * <ActionBar title='Cool title' />
 * ```
 */
export const ActionBar = memo(
  ({
    title,
    'data-qa': dataQa,
    shouldDisableButton = false,
    shouldHaveButton = false,
    buttonAction,
    buttonLabel,
    typographyVariant = 'h1',
    customclasses
  }: ActionBarProps) => {
    const getTitle = useMemo<ActionBarProps['title']>(() => {
      if (typeof title === 'string') {
        return (
          <Typography
            variant={typographyVariant}
            data-qa='action-bar-title'
            color='textPrimary'
            className={classnames(css.actionHeader, customclasses)}
          >
            {title}
          </Typography>
        );
      }

      if (typeof title === 'function') return title();

      return title;
    }, [customclasses, title, typographyVariant]);

    return (
      <Box component='div' className={classnames(css.actionBar)} data-qa={dataQa}>
        <Grid container direction='row' justify='space-between' alignItems='center' className={css.actionGridLeft}>
          <Grid item key={1} xs={6}>
            {getTitle}
          </Grid>
          <When condition={Boolean(shouldHaveButton)}>
            <Grid item key={2} xs={5} sm={6} className={css.actionGridRight}>
              <Button
                data-qa='action-bar-button'
                variant='contained'
                color='primary'
                label={buttonLabel}
                onClick={() => (buttonAction ? buttonAction() : null)}
                disabled={Boolean(shouldDisableButton)}
              />
            </Grid>
          </When>
        </Grid>
      </Box>
    );
  }
);

export default ActionBar;
