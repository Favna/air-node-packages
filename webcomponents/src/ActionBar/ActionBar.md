### Usage of the ActionBar component:

**Examples:**

```tsx
import React from 'react';

import { ActionBar } from '@rws-air/webcomponents';

<ActionBar title='ActionBar Title' buttonLabel='action-bar-button' buttonAction={() => console.log('test')} shouldHaveButton />
```

```tsx
import React from 'react';

import { ActionBar } from '@rws-air/webcomponents';

<ActionBar title='ActionBar Title' />
```