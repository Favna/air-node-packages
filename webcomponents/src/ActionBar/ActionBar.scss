@import '../styles/vars';

.actionBar {
  background-color: $light-blue !important;
  height: 5rem !important;
  line-height: 5rem !important;
}

.actionBarHeader {
  color: $text-black !important;
  font-size: 2.2rem !important;
  margin: 0 !important;
  padding-left: 26px !important;
}

@media only screen and (max-width: 1024px) {
  .actionBarHeader {
    font-size: 1.6rem !important;
  }
}

@media only screen and (max-width: 800px) {
  .actionBarHeader {
    font-size: 1.2rem !important;
  }
}
