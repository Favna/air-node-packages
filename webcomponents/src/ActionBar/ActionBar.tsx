import Grid from '@material-ui/core/Grid/Grid';
import Button from '../Button/Button';
import classnames from 'classnames';
import React, { FC, Fragment } from 'react';
import css from './ActionBar.scss';

export type ActionBarProps = {
    /** The title to show in the ActionBar */
    title: string;
    /** Whether an action button should be displayed in the ActionBar */
    shouldHaveButton?: boolean;
    /** The text that should be shown in the button */
    buttonLabel?: string;
    /** The action that should be invoked when the button is clicked */
    buttonAction?: (...args: any[]) => any;
};

const ActionBar: FC<ActionBarProps> = props => {
    return (
        <div className={classnames('navigation-bar', css.actionBar)}>
            <Grid container direction='row' justify='space-between' alignItems='center' style={{ height: '100%' }}>
                <Grid item key={1} xs={6}>
                    <h1 data-qa='action-bar-title' className={classnames(css.actionBarHeader)}>
                        {props.title}
                    </h1>
                </Grid>
                <Grid item key={2} xs={6}>
                    {props.shouldHaveButton ?
                        <Button
                            data-qa='action-bar-button'
                            variant='contained'
                            color='primary'
                            label={props.buttonLabel}
                            onClick={() => props.buttonAction ? props.buttonAction() : null}
                        />
                        : <Fragment />
                    }
                </Grid>
            </Grid>
        </div>
    );
};

export default ActionBar;
