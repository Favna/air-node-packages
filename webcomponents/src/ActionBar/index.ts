export { default as ActionBar } from './ActionBar';
export { ActionBarProps } from './ActionBar';