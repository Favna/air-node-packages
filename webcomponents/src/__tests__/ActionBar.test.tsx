import ActionBar from '../ActionBar/ActionBar';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

const mockCallback = jest.fn();
let actionBar: ShallowWrapper;

test('should match snapshot', () => {
  actionBar = shallow(<ActionBar title='42' buttonLabel='label' buttonAction={mockCallback} shouldHaveButton />);
  expect(actionBar).toMatchSnapshot();
});

describe('ActionBar with button', () => {
  test('confirm the button click is registered', () => {
    actionBar = shallow(<ActionBar title='42' buttonLabel='label' buttonAction={mockCallback} shouldHaveButton />);
    const actionButton = actionBar.find('[data-qa="action-bar-button"]');
    actionButton.simulate('click');
    expect(mockCallback).toHaveBeenCalled();
    expect(actionButton.dive().text()).toBe('label');
  });

  test('confirm the button click passes arguments', () => {
    const param = 42;

    actionBar = shallow(
      <ActionBar title='42' buttonLabel='label'
        buttonAction={() => mockCallback(param)}
        shouldHaveButton
      />
    );

    const actionButton = actionBar.find('[data-qa="action-bar-button"]');
    actionButton.simulate('click');
    expect(mockCallback).toHaveBeenCalled();
    expect(mockCallback).toHaveBeenCalledWith(param);
    expect(actionButton.dive().text()).toBe('label');
  });
});

describe('ActionBar without button', () => {
  beforeEach(() => actionBar = shallow(<ActionBar title='42' />));

  test('should show title', () => {
    const title = actionBar.find('[data-qa="action-bar-title"]');
    const actionButton = actionBar.find('[data-qa="action-bar-button"]');
    expect(title.text()).toBe('42');
    expect(actionButton.length).toBe(0);
  });
});

describe('Snapshot Testing', () => {
  test('Required Props', () => {
    const actionBar = shallow(<ActionBar title='42' />);
    expect(actionBar).toMatchSnapshot();
  });

  test('With Button', () => {
    const actionBar = shallow(
      <ActionBar
        title='42' buttonLabel='label'
        buttonAction={mockCallback} shouldHaveButton
      />
    );
    expect(actionBar).toMatchSnapshot();
  });

  test('AdditionalTest', () => {
    const actionBar = shallow(
      <ActionBar
        title='42' buttonLabel='label'
        buttonAction={mockCallback} shouldHaveButton
        customclasses={'snapshot-class'} data-qa='snapshot-qa'
      />
    );
    expect(actionBar).toMatchSnapshot();
  });
});