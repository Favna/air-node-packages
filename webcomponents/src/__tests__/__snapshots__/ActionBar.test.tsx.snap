// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="navigation-bar"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    direction="row"
    justify="space-between"
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="1"
      xs={6}
    >
      <h1
        className=""
        data-qa="action-bar-title"
      >
        42
      </h1>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="2"
      xs={6}
    >
      <Button
        color="primary"
        data-qa="action-bar-button"
        label="label"
        onClick={[Function]}
        variant="contained"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
